### ISSUE TITLE
Keyboard Controllers 

### ISSUE THREAD

**Issue by CaptMoore** [DirtyHairy/r77-firmware-ng#9]
I tried using the keyboard adapters (CX50) with the Retron77, and I couldn't get them to work, either with the 2600-adapter, or using the build in port.  The built in port simply doesn't work at all, with the 2600-adapter, a couple of keys would bring up the options menu, but they mostly didn't work (and didn't do anything in game.)

As a test, I did then go to the macOS version of stella, using the same controllers and they work perfectly with the 2600-adapter (D9 version).

Could someone else confirm my findings, and look into allowing the keyboard controller to work with the retron 77 version of stella?

**Comment by CaptMoore** [comment]
I have done a little more troubleshooting with the keyboard controllers.  Setup is simply one 2600-adapter (D9), set up in keyboard mode by pressing 0 once the usb is powered up.  I am testing this using atari's codebreaker.

Numbers 1/3 work normally.  Number 4 brings up the command menu.  Number 5 goes to the game select manual.  Number 6 goes the setting menu.  Number 7 pauses stella. Number 8 activates the select key.  Number 9 is the reset key.  The star key appears to work normally.

Basically, I believe that the keypad is activating buttons that have been remapped to different part of the emulators (perhaps the buttons build into the console?)

I have tried, but haven't had much luck remapping the controls. I am able to set the button assignments, but no luck so far actually getting them to stop pulling up various stella menus.

**Comment by thrust26** [comment]
The console buttons are mapped to function keys. So they should not interfere here.

I haven't tested R77 with keyboard controllers (not available to me) yet, So it may take quite some time until someone comes back to this.

**Comment by CaptMoore** [comment]
I loaded the fresh image of the retron 77 Stella 6.6 today.

It looks like this issues is fixed, keyboard controllers work normally with the 2600-daptor D9.  I was able to use Paul Slocum's Test cart rom to verify both keyboard controllers are working normally with the latest firmware.

Link to test App: https://atariage.com/2600/roms/Testcart.zip

This issue was reported as fixed by Thomas Jentzsch to me in a prerelease version discussion.

Link to forum post:
https://atariage.com/forums/topic/327250-stella-66-released/?do=findComment&comment=4952718

It looks like this bug can be knocked off the list.

> The console buttons are mapped to function keys. So they should not interfere here.
> 
> I haven't tested R77 with keyboard controllers (not available to me) yet, So it may take quite some time until someone comes back to this.
